Core/Sources/Configuration/GoogleCloudCredentialsConfiguration.swift

```diff
@@ -24,7 +24,7 @@ public struct GoogleCloudCredentialsConfiguration {
         // https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application
         let serviceAccountCredentialsPath = env["GOOGLE_APPLICATION_CREDENTIALS"] ??
                                              credentialsFile ??
-                                             "~/.config/gcloud/application_default_credentials.json"
+                                             "\(NSHomeDirectory())/.config/gcloud/application_default_credentials.json"
```

Comment:
@SusanDoggie Just want to verify that this will compile and work on linux?

Reply:
It will be compiled but I don’t know is it always correct for Linux.

Reference

https://github.com/apple/swift-corelibs-foundation/blob/2a5bc4d8a0b073532e60410682f5eb8f00144870/Sources/Foundation/NSPathUtilities.swift#L710

https://github.com/apple/swift-corelibs-foundation/blob/791d57e942fbe3f163f4162ed5fc0021e9f7f39a/CoreFoundation/Base.subproj/CFPlatform.c#L396

Reply:
Can you verify that this gives you the expected result on Linux? Maybe run it in a docker container to make sure nothing breaks. @SusanDoggie

Reply:
I have tested in docker and it works fine.

```
print("\(NSHomeDirectory())/.config/gcloud/application_default_credentials.json")
```
will get the result
```
/root/.config/gcloud/application_default_credentials.json
```

which is default location that the command `gcloud auth application-default login` install credential.

Reply:
Awesome!